### PR TITLE
Introduce OpenShift metadata to generated resources

### PR DESCRIPTION
**Pull request by lyarwood**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
OpenShift specific metadata has been introduced to all `VirtualMachinePreference` and `VirtualMachineClusterPreference` resources.

A new `instancetype.kubevirt.io/os-type` label has also been introduced allowing users to query for preferences that match the underlying OS type of their workload.
```

### CODE REVIEW

File: common-clusterpreferences-bundle.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 apiVersion: instancetype.kubevirt.io/v1alpha2
 kind: VirtualMachineClusterPreference
 metadata:
+  annotations:
+    openshift.io/documentation-url: https://github.com/kubevirt/common-instancetypes
+    openshift.io/provider-display-name: KubeVirt
```
Comment (0xFelix) [inline common-clusterpreferences-bundle.yaml:7]: Is the an appropriate icon for alpine? Maybe a generic one?

File: common-clusterpreferences-bundle.yaml [hunk 2]
```diff
@@ -49,6 +74,13 @@ spec:
 apiVersion: instancetype.kubevirt.io/v1alpha2
 kind: VirtualMachineClusterPreference
 metadata:
+  annotations:
+    iconClass: icon-centos
+    openshift.io/display-name: CentOS Stream 8
+    openshift.io/documentation-url: https://github.com/kubevirt/common-instancetypes
+    openshift.io/provider-display-name: KubeVirt
+    openshift.io/support-url: https://github.com/kubevirt/common-instancetypes/issues
+    tags: hidden,kubevirt,linux,centos
```
Comment (0xFelix) [inline common-clusterpreferences-bundle.yaml:83]: Change tag `centos` to `centos-stream`? Or add tag `stream`?

File: common-clusterpreferences-bundle.yaml [hunk 3]
```diff
@@ -2,6 +2,10 @@
 apiVersion: instancetype.kubevirt.io/v1alpha2
 kind: VirtualMachineClusterPreference
 metadata:
+  annotations:
+    openshift.io/documentation-url: https://github.com/kubevirt/common-instancetypes
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:6]: I wasn't going to add display-name etc to unsupported OSes downstream. I can but they will only be dropped from the bundles we ship there anyway.

File: common-clusterpreferences-bundle.yaml (hunk 1 above)
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:7]: As above, I doubt we will ship this downstream so I didn't want to add an icon here.

File: common-clusterpreferences-bundle.yaml [hunk 4]
```diff
@@ -2,6 +2,10 @@
...
+    openshift.io/provider-display-name: KubeVirt
+    openshift.io/support-url: https://github.com/kubevirt/common-instancetypes/issues
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:8]: As above.

File: common-clusterpreferences-bundle.yaml [hunk 5]
```diff
@@ -102,6 +148,10 @@ spec:
 apiVersion: instancetype.kubevirt.io/v1alpha2
 kind: VirtualMachineClusterPreference
 metadata:
+  annotations:
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:151]: As above.

File: common-clusterpreferences-bundle.yaml [hunk 6]
```diff
@@ -220,6 +316,10 @@ spec:
...
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:319]: As above.

File: common-clusterpreferences-bundle.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (0xFelix) [reply common-clusterpreferences-bundle.yaml:6]: What about OKD? Can't it be used there?

File: common-clusterpreferences-bundle.yaml (hunk 1 above)
Comment (0xFelix) [reply common-clusterpreferences-bundle.yaml:7]: What about OKD? Can't it be used there?

File: common-clusterpreferences-bundle.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lyarwood) [reply common-clusterpreferences-bundle.yaml:6]: Until I saw https://github.com/kubevirt-ui/kubevirt-plugin/pull/1025 I was going to suggest we didn't need to worry about OKD but it looks like the UI is usable upstream so I'll add some metadata for these preferences as well.

File: common-preferences-bundle.yaml [hunk 7]
```diff
@@ -232,6 +344,13 @@ spec:
 apiVersion: instancetype.kubevirt.io/v1alpha2
 kind: VirtualMachinePreference
 metadata:
+  annotations:
+    iconClass: icon-windows
+    openshift.io/display-name: Microsoft Windows 10
```
Comment (0xFelix) [inline common-preferences-bundle.yaml:349]: Would it make sense to be able to distinguish the regular windows preferences and the windows virtio preferences by their display name?

Comment (lyarwood) [reply common-preferences-bundle.yaml:349]: True, added thanks.

**Comment by lyarwood** [comment]
I've also added an `instancetype.kubevirt.io/os-type` label to allow users a quick way to query for preferences given the underlying type of OS supported. There are likely more labels we can add here so suggestions welcome!

**Comment by 0xFelix** [comment]
Thanks!

**Pull request merged**